### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by JamesOlvertone** [zlatinb/muwire#137]
Scenario:
You search for something, have some results, click one user and youl get some kind of folder structure where this match occured. Maybe the files, or Foldername whatever.
Now you want see more of the subfolder. You browse the users share and it loads ALL its shared files. If he shares a lot of files it takes forever to load all. 
New Feature. Browse only the subfolder shown in search results. -> Browsing is faster and risk that something is lost or gets stuck (see my other bug report) is much lower.

Thank you.

**Comment by zlatinb** [comment]
I agree this is a very useful feature, but unfortunately it's pretty difficult to implement.  I'll think about it but I don't promise anything.
